### DISCUSSION TITLE
How access navigatorkey  to navigate throw Firebase messaging

### DISCUSSION THREAD

**Discussion by felixferr** [csells/go_router#220] [Q&A]
Hi, i would like to use navigatorkey to handle navigation in firebase messagng . if it's not possible, what is the solution ?

**Comment by csells** [comment]
you should be able to map navigatorkey to a route by name or path as you choose

**Reply by felixferr** [reply]
Im not shure to understand. Have you got an example please ? I initialize firebase messaging and the streamsubscription on the initstate of Myapp. So i don't have access to Gorouter
